src/fastcs/transports/tango/transport.py

```
@@ -15,10 +16,23 @@ class TangoTransport(Transport):
 
     def connect(
         self,
-        controller_api: ControllerAPI,
+        controller_apis: list[ControllerAPI],
         loop: asyncio.AbstractEventLoop,
     ):
-        self._dsr = TangoDSR(controller_api, loop)
+        seen: dict[str, str] = {}
+        for api in controller_apis:
+            id = api.path[0]
```

Ruff (0.15.13)

[error] Variable `id` is shadowing a Python builtin

(A001)